aquadoggo/migrations/20230811201924_purge-document-trigger.sql

```diff
@@ -0,0 +1,14 @@
+-- SPDX-License-Identifier: AGPL-3.0-or-later
+
+CREATE TRIGGER purge_document 
+   AFTER DELETE ON documents
+BEGIN
+    DELETE FROM entries 
+    WHERE entries.entry_hash IN (
+        SELECT operations_v1.operation_id FROM operations_v1
+        WHERE operations_v1.document_id = OLD.document_id
+    );
+    
+    DELETE FROM operations_v1
+    WHERE operations_v1.document_id = OLD.document_id;
+END;
```

Comment:
This was the part which was not supported by PostgreSQL, as we said in our last meeting, lets move it into separate SQL queries in Rust to allow compatibility (for now ..).